lambdas/models/pds_models.py

```diff
@@ -80,41 +83,57 @@ def get_security(self) -> Security:
 
     def is_unrestricted(self) -> bool:
         security = self.get_security()
-        if security.code == "U":
-            return True
-        return False
+        return security.code == "U"
 
-    def get_current_usual_name(self) -> [Optional[Name]]:
+    def get_current_usual_name(self) -> Optional[Name]:
         for entry in self.name:
             if entry.use.lower() == "usual":
                 return entry
 
+    def get_current_family_name_and_given_name(self) -> Tuple[str, list[str]]:
+        usual_name = self.get_current_usual_name()
+        if not usual_name:
+            logger.warning("The patient does not have a usual name.")
+            return "", [""]
+        given_name = usual_name.given
+        family_name = usual_name.family
+
+        if not given_name or given_name == [""]:
+            logger.warning("The given name of patient is empty.")
+
+        return family_name, given_name
+
     def get_current_home_address(self) -> Optional[Address]:
         if self.is_unrestricted() and self.address:
             for entry in self.address:
                 if entry.use.lower() == "home":
                     return entry
 
     def get_active_ods_code_for_gp(self) -> str:
-        if self.general_practitioner:
-            for entry in self.general_practitioner:
-                gp_end_date = entry.identifier.period.end
-                if not gp_end_date or gp_end_date >= date.today():
-                    return entry.identifier.value
+        for entry in self.general_practitioner:
+            period = entry.identifier.period
+            if not period:
+                continue
+            gp_end_date = period.end
+            if not gp_end_date or gp_end_date >= date.today():
+                return entry.identifier.value
         return ""
 
     def get_is_active_status(self) -> bool:
         gp_ods = self.get_active_ods_code_for_gp()
         return bool(gp_ods)
 
     def get_patient_details(self, nhs_number) -> PatientDetails:
+        family_name, given_name = self.get_current_family_name_and_given_name()
+        current_home_address = self.get_current_home_address()
+
         patient_details = PatientDetails(
-            givenName=self.get_current_usual_name().given,
-            familyName=self.get_current_usual_name().family,
+            givenName=given_name,
+            familyName=family_name,
             birthDate=self.birth_date,
             postalCode=(
-                self.get_current_home_address().postal_code
-                if self.is_unrestricted() and self.address
+                current_home_address.postal_code
+                if self.is_unrestricted() and current_home_address
```

Comment:
We're already checking `self.is_unrestricted()` in `self.get_current_home_address()`

Reply:
Thanks!  I've removed the duplicated check here